internal/common/connectionsapi/client.go

```diff
@@ -0,0 +1,144 @@
+package connectionsapi
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+type Client struct {
+	authToken string
+	apiURL    url.URL
+	client    *http.Client
+}
+
+const (
+	defaultRetries = 3
+	defaultTimeout = 90 * time.Second
+)
+
+func NewClient(authToken string, rawAPIURL string, client *http.Client) (*Client, error) {
```

Comment:
nit: Maybe we can just call this `url`? It should be clear it's for the API as the caller is creating a client for it

Reply:
Alright, I changed it to `rawURL` as there is a clash with the import name. Let me know if you prefer something else.